packages/compiler/test/formatter/formatter.test.ts

```diff
@@ -2429,6 +2429,41 @@ model Foo {
     });
   });
 
+  describe("valueof", () => {
+    it("format simple valueof", async () => {
+      await assertFormat({
+        code: `
+alias A =      valueof        string;
+`,
+        expected: `
+alias A = valueof string;
+`,
+      });
+    });
+
+    it("keeps parentheses around valueof inside a union", async () => {
+      await assertFormat({
+        code: `
+alias A =      (valueof        string) | Model;
+`,
+        expected: `
+alias A = (valueof string) | Model;
+`,
+      });
+    });
+
+    it("keeps parentheses around valueof inside a array expression", async () => {
```

Comment:
We should have a test for intersection as well.  Otherwise, lgtm

Reply:
I added intersection in the condition because technically it is ahead but you also cannot intersect non model so not sure it is very valuable